pkg/build/build.go

```diff
@@ -23,8 +23,10 @@ import (
 type Interface interface {
 	// IsSupportedReference determines whether the given reference is to an importpath reference
 	// that Ko supports building.
+	// `tests` activates 'test-builds'. So importpaths that do not provide `package main` but provide tests (*_test.go).
 	// TODO(mattmoor): Verify that some base repo: foo.io/bar can be suffixed with this reference and parsed.
-	IsSupportedReference(string) bool
+
+	IsSupportedReference(ip string, tests bool) bool
```

Comment:
Instead of a flag we could think of not returning bool but more detailed information what kind of reference it is.
eg.

type ReferenceType string

const (
    MAIN ReferenceType = "main"
    TEST ReferenceType = "test"
    NONE ReferenceType = "none"
)
Or - in order to remove the logic from the caller - we could just allow the caller to specify what he needs
e.g.

type AllowedReferences struct {
	Test bool
	Main bool
}

type Option func(*AllowedReferences)

func AllowTest() Option {
	return func(a *AllowedReferences) {
		a.Test = true
	}
}

func AllowMain() Option {
	return func(a *AllowedReferences) {
		a.Main = true
	}
}

func NewAllowedReferences(o ...Option) *AllowedReferences {
	ref := &AllowedReferences{}
	for _, a := range o {
		a(ref)
	}
	return ref
}
And add this to issupportedreference